### PR TITLE
🐛 [RUMF-1124] Use the programmatic action attribute instead of innertext

### PR DESCRIPTION
**Pull request by MaheshGit233**
🐛 Use the programmatic action attribute instead of innertext for fetching the textual content

## Motivation

When fetching the textual content programmatic are not being considered which may log user sensitive data. Replacing the programmatic attribute value with innertext will help in avoiding this.

## Changes

Replace the programmatic attribute value if the element has the attribute, otherwise use the innertext in getTextualContent function. This change will help us in scrubbing the data (inside the elements which are not clickable) which may have user information.

## Testing

Local
Unit

Verified the changes in local and added the unit test cases for the changes.

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

**Comment by MaheshGit233** [comment]
@DataDog/rum-browser Please review the above changes.

File: packages/rum-core/src/domain/rumEventsCollection/action/getActionNameFromElement.ts [hunk 1]
```diff
@@ -175,6 +175,18 @@ function getTextualContent(element: Element | HTMLElement) {
...
     }
+	
+	const elementsTextToScrub: NodeListOf<HTMLElement> = element.querySelectorAll(`[${DEFAULT_PROGRAMMATIC_ATTRIBUTE}]`)
+
+    // replace the innertext of the elements with programmatic attribute value
+    for (let i = 0; i < elementsTextToScrub.length; i += 1) {
+      const innerText = elementsTextToScrub[i].innerText
+      // eslint-disable-next-line @typescript-eslint/prefer-for-of
+      if (innerText.trim().length > 0) {
+        text = text.replace(innerText, elementsTextToScrub[i].getAttribute(`${DEFAULT_PROGRAMMATIC_ATTRIBUTE}`) || '')
```
Comment (BenoitZugmeyer) [inline packages/rum-core/src/domain/rumEventsCollection/action/getActionNameFromElement.ts:186]:
```suggestion
        text = text.replace(innerText, elementsTextToScrub[i].getAttribute(DEFAULT_PROGRAMMATIC_ATTRIBUTE) || '')
```

File: packages/rum-core/src/domain/rumEventsCollection/action/getActionNameFromElement.ts [hunk 2]
```diff
@@ -175,6 +175,18 @@ function getTextualContent(element: Element | HTMLElement) {
         }
       }
     }
+	
+	const elementsTextToScrub: NodeListOf<HTMLElement> = element.querySelectorAll(`[${DEFAULT_PROGRAMMATIC_ATTRIBUTE}]`)
+
+    // replace the innertext of the elements with programmatic attribute value
+    for (let i = 0; i < elementsTextToScrub.length; i += 1) {
+      const innerText = elementsTextToScrub[i].innerText
+      // eslint-disable-next-line @typescript-eslint/prefer-for-of
```
Comment (BenoitZugmeyer) [inline packages/rum-core/src/domain/rumEventsCollection/action/getActionNameFromElement.ts:184]: Could you use `elementsTextToScrub.forEach`?

**Review by BenoitZugmeyer — CHANGES_REQUESTED** [review]
Hello @MaheshGit233 . Your PR looks great, thank you. But it is a breaking change: action names will be changed when upgrading the SDK.

Luckily, we are in the process of releasing a new major version! So I'll make sure your PR is shiped with the v4 of the Browser SDK.

File: packages/rum-core/src/domain/rumEventsCollection/action/getActionNameFromElement.ts (hunk 1 above)
Comment (MaheshGit233) [reply packages/rum-core/src/domain/rumEventsCollection/action/getActionNameFromElement.ts]: Addressed.. Taken the suggestion

File: packages/rum-core/src/domain/rumEventsCollection/action/getActionNameFromElement.ts (hunk 2 above)
Comment (MaheshGit233) [reply packages/rum-core/src/domain/rumEventsCollection/action/getActionNameFromElement.ts:184]: Addressed. Replaced for with foreach

File: packages/rum-core/src/domain/rumEventsCollection/action/getActionNameFromElement.ts (hunk 1 above)
Comment (MaheshGit233) [reply packages/rum-core/src/domain/rumEventsCollection/action/getActionNameFromElement.ts]: Thank you for reviewing the changes. Please let me know if you have any comments/suggestions

File: packages/rum-core/src/domain/rumEventsCollection/action/getActionNameFromElement.spec.ts [hunk 3]
```diff
@@ -27,6 +27,18 @@ describe('getActionNameFromElement', () => {
...
+      'Foo custom action'
+    )
+  })
+
+  it('replaces the programmatic action name in textual content based on the user-defined attribute', () => {
+    expect(
+      getActionNameFromElement(element`<div>Foo <div data-test-id="custom action">bar<div></div>`, 'data-test-id')
+    ).toBe('Foo custom action')
+  })
+
```
Comment (bcaudan) [inline packages/rum-core/src/domain/rumEventsCollection/action/getActionNameFromElement.spec.ts:41]: nit: for consistency with the file organization, those tests should be in the `programmatically declared action name` suite

**Review by bcaudan — APPROVED** [review]

**Comment by BenoitZugmeyer** [comment]
@MaheshGit233 again, thank you for your contribution! This will be released with the v4 release, in early january.

**Comment by MaheshGit233** [comment]
Thank you for considering the changes @BenoitZugmeyer

**Pull request merged**